### PR TITLE
Added STM32L432KC DAC support.

### PR DESCRIPTION
**Pull request by daniel-p-carvalho**
## Summary

This PR adds support to STM32L432KC internal DAC on nucleo-l432kc board.

## Testing

The changes was tested using DAC Tool.

### CODE REVIEW

File: boards/arm/stm32l4/nucleo-l432kc/src/stm32_dac.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/*****************************************************************************
+ * boards/arm/stm32l4/nucleo-l432kc/src/stm32_dac.c
+ *
+ *   Copyright (C) 2017 Haltian Ltd. All rights reserved.
```
Comment (xiaoxiang781216) [inline boards/arm/stm32l4/nucleo-l432kc/src/stm32_dac.c:4]: @daniel-p-carvalho could you create a new patch to change the copright to Apache?

Comment (daniel-p-carvalho) [reply boards/arm/stm32l4/nucleo-l432kc/src/stm32_dac.c:4]: @xiaoxiang781216 this file was copied from nucleo-l452re so I think that I don't have the rights to do the copyright change unless Haltian Ltd. give the permission.

Comment (daniel-p-carvalho) [reply boards/arm/stm32l4/nucleo-l432kc/src/stm32_dac.c:4]: @juniskane did you signed the ICLA or Haltian signed the SGA?

Comment (patacongo) [reply boards/arm/stm32l4/nucleo-l432kc/src/stm32_dac.c:4]: I think that this change must be reverted.  We cannot being ANY new files into the repositories that do not have Apache 2.0 license.

UNLESS we follow the rules for third party software.  If it is not Apache 2.0, it is third party and not truly a part of NuttX.

**Comment by daniel-p-carvalho** [comment]
@patacongo I've created the STM32L4 common driver directory trying to replicate the one in STM32 boards directory. It builds and seems to work fine but the build process is showing the following messages

make[2]: Entering directory '/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/arm/stm32l4/common'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:102: warning: overriding recipe for target 'stm32_boot.o'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:102: warning: ignoring old recipe for target 'stm32_boot.o'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:102: warning: overriding recipe for target 'stm32_spi.o'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:102: warning: ignoring old recipe for target 'stm32_spi.o'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:102: warning: overriding recipe for target 'stm32_autoleds.o'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:102: warning: ignoring old recipe for target 'stm32_autoleds.o'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:102: warning: overriding recipe for target 'stm32_appinit.o'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:102: warning: ignoring old recipe for target 'stm32_appinit.o'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:108: warning: overriding recipe for target 'libboard.a'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:108: warning: ignoring old recipe for target 'libboard.a'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:114: warning: overriding recipe for target '.depend'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:114: warning: ignoring old recipe for target '.depend'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:129: warning: overriding recipe for target 'clean'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:129: warning: ignoring old recipe for target 'clean'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:136: warning: overriding recipe for target 'distclean'
/home/daniel/nuttx-workspace/stm32l4_adc_llops/nuttx/boards/Board.mk:136: warning: ignoring old recipe for target 'distclean'

Can you help me with this problem?

Thanks

**Pull request merged**